### PR TITLE
Add / fix smithy-client docs

### PR DESCRIPTION
**Pull request by rcoh**
## Motivation and Context
Improve docs for constructing smithy clients

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: rust-runtime/aws-smithy-client/src/hyper_impls.rs [hunk 1]
```diff
@@ -122,6 +122,24 @@ fn find_source<'a, E: Error + 'static>(err: &'a (dyn Error + 'static)) -> Option
 
 #[derive(Default, Debug)]
 /// Builder for [`HyperAdapter`]
+///
```
Comment (jdisanti) [inline rust-runtime/aws-smithy-client/src/hyper_impls.rs:125]: Should this doc comment go on HyperAdapter itself? At least when I was digging into the docs, that's where I landed first, and then made my way into the builder doc once I knew I wanted to use it and needed to construct it.

**Review by paholg — APPROVED** [review]
Thanks, this would have helped me!

File: rust-runtime/aws-smithy-client/src/hyper_impls.rs [hunk 2]
```diff
@@ -122,6 +122,24 @@ fn find_source<'a, E: Error + 'static>(err: &'a (dyn Error + 'static)) -> Option
...
+/// Construct a HyperAdapter with the default HTTP implementation (rustls). This can be useful when you want to share a Hyper connector
+/// between multiple Smithy clients.
+///
+/// ```rust
+/// use tower::layer::util::Identity;
+/// use aws_smithy_client::{conns, hyper_ext};
+/// use aws_smithy_client::erase::DynConnector;
+/// let hyper_connector = hyper_ext::Adapter::builder().build(conns::https());
+/// // this client can then be used when constructing a Smithy Client
+/// // NOTE: in real usage, you will almost certainly want a real middleware implementation
```
Comment (jdisanti) [inline rust-runtime/aws-smithy-client/src/hyper_impls.rs:140]: What do you mean by this note?

File: rust-runtime/aws-smithy-client/src/hyper_impls.rs [hunk 3]
```diff
@@ -122,6 +122,24 @@ fn find_source<'a, E: Error + 'static>(err: &'a (dyn Error + 'static)) -> Option
...
+///
+/// # Examples
+/// Construct a HyperAdapter with the default HTTP implementation (rustls). This can be useful when you want to share a Hyper connector
+/// between multiple Smithy clients.
+///
+/// ```rust
+/// use tower::layer::util::Identity;
+/// use aws_smithy_client::{conns, hyper_ext};
+/// use aws_smithy_client::erase::DynConnector;
+/// let hyper_connector = hyper_ext::Adapter::builder().build(conns::https());
```
Comment (jdisanti) [inline rust-runtime/aws-smithy-client/src/hyper_impls.rs:138]:
```suggestion
/// use aws_smithy_client::erase::DynConnector;
///
/// let hyper_connector = hyper_ext::Adapter::builder().build(conns::https());
```

File: rust-runtime/aws-smithy-client/src/hyper_ext.rs [hunk 4]
```diff
@@ -14,22 +49,22 @@ use tower::{BoxError, Service};
...
-/// Adapter from a [`hyper::Client`] to a connector usable by a [`Client`](crate::Client).
+/// Adapter from a [`hyper::Client`](hyper::Client) to a connector usable by a Smithy [`Client`](crate::Client).
 ///
-/// This adapter also enables TCP connect and HTTP read timeouts via [`HyperAdapter::builder`]
+/// This adapter also enables TCP `CONNECT` and HTTP `READ` timeouts via [`Adapter::builder`]. For examples
+/// see [the module documentation](crate::hyper_ext).
 #[derive(Clone, Debug)]
 #[non_exhaustive]
-pub struct HyperAdapter<C>(HttpReadTimeout<hyper::Client<ConnectTimeout<C>, SdkBody>>);
+pub struct Adapter<C>(HttpReadTimeout<hyper::Client<ConnectTimeout<C>, SdkBody>>);
```
[the remark this answers is not in the document]
Comment (rcoh) [reply rust-runtime/aws-smithy-client/src/hyper_ext.rs:65]: no, we were publicly exporting it as `Adapter`

**Review by jdisanti — APPROVED** [review]

**Pull request merged**